### PR TITLE
#74: Fix DATEADD tests

### LINKED ISSUE #74: TDVT tests are failing
The TDVT tests for JDBC connector fail with the following error message in `tdvt_log_combined.txt`:

```
2023-12-11 07:55:34,544 Thread-[1] Running test:logical.calcs.exasol_jdbc
2023-12-11 07:55:34,553 Thread-[1]  calling C:\Program Files\Tableau\Tableau 2023.3\bin\tabquerytool.exe --query-file-list C:\Users\chp\AppData\Local\Temp\2\cca01878f059c4e8bf387d353dba092ed4fa2b543a98fa40ee74a993a3q6rpt0\logical_calcs_exasol_jdbc\tests.txt -d C:\Users\chp\git\tableau-connector\tdvt_jdbc\tds\cast_calcs.exasol_jdbc.tds --combined --schema TestV1 --password-file C:\Users\chp\git\tableau-connector\tdvt_jdbc\tds\exasol_jdbc.password --output-dir C:\Users\chp\AppData\Local\Temp\2\cca01878f059c4e8bf387d353dba092ed4fa2b543a98fa40ee74a993a3q6rpt0 -DLogDir=C:\Users\chp\AppData\Local\Temp\2\cca01878f059c4e8bf387d353dba092ed4fa2b543a98fa40ee74a993a3q6rpt0\logical_calcs_exasol_jdbc -DOverride=ProtocolServerNewLog -DConnectPluginsPath=..\src -DLogicalQueryRewriteDisable=Funcall:RewriteConstantFuncall -DInMemoryLogicalCacheDisable
2023-12-11 07:55:47,401 Thread-[1] CalledProcessError: Return code: 12 Attempting to run query...
TableauException: FederatedSubtreeBuilder: Could not determine types for all columns in the relation.
{"relation-conn":"leaf","relation-conn-class":"exasol_jdbc","relation-table":"[TESTV1].[Calcs]","unknown-type-columns":["[bool0_]","[bool1_]","[bool2_]","[bool3_]"]}

Error running query

2023-12-11 07:55:47,402 Thread-[1] Command line output:Attempting to run query...
TableauException: FederatedSubtreeBuilder: Could not determine types for all columns in the relation.
{"relation-conn":"leaf","relation-conn-class":"exasol_jdbc","relation-table":"[TESTV1].[Calcs]","unknown-type-columns":["[bool0_]","[bool1_]","[bool2_]","[bool3_]"]}

Error running query

2023-12-11 07:55:47,412 
Finished tdvt suite [exasol_jdbc]: tested sql [False]: tested tuples [True]: tested error [False]: output dir [C:\Users\chp\AppData\Local\Temp\2\cca01878f059c4e8bf387d353dba092ed4fa2b543a98fa40ee74a993a3q6rpt0]: logical [True]: config file [logical.calcs.exasol_jdbc]: override [-DConnectPluginsPath=..\src]: tds [C:\Users\chp\git\tableau-connector\tdvt_jdbc\tds\cast_calcs.exasol_jdbc.tds]: thread [6]
```

In total 31 tests fail because of this:
```
Test Count: 949 tests
        Passed tests: 918
        Failed tests: 31
        Tests run: 949
        Disabled tests: 0
        Skipped tests: 0
```

TDVT version: 2.7.6
Tableau Desktop version: 2023.3

See question in TDVT repo: https://github.com/tableau/connector-plugin-sdk/issues/1206

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #74

### CODE REVIEW

File: javascript-test/package-lock.json
Comment (ckunki) [inline javascript-test/package-lock.json:1]: Could me mark this file as "generated"?

File: tools/package_connector.sh [hunk 1]
```diff
@@ -3,12 +3,10 @@ set -euo pipefail
 
 project_dir="$( cd "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P )"
 readonly project_dir
```
Comment (ckunki) [inline tools/package_connector.sh:5]: Is there a specific reason for the other variables to be 1-liners while `project_dir` is defined and made read-only in separate lines?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: javascript-test/package-lock.json
Comment (kaklakariada) [reply javascript-test/package-lock.json:1]: It is already marked as generated:

```
javascript-test/package-lock.json           linguist-generated=true
```

File: tools/package_connector.sh (hunk 1 above)
Comment (kaklakariada) [reply tools/package_connector.sh:5]: I remember that the `shellcheck` tool complains if you do this in one line 🤷

Comment (ckunki) [reply tools/package_connector.sh:5]: Yes, I think so, too, but the other vars seem to be OK. Maybe simpler expressions?

File: javascript-test/package-lock.json
Comment (ckunki) [reply javascript-test/package-lock.json:1]: Strange. Still GitHub displays only "Load diff" and not "...generated".
Could it be that file `.gitattributes` requires a trailing newline?

**Review by ckunki — APPROVED** [review]

**Pull request merged**